controllers/courses.js

```diff
@@ -609,13 +617,19 @@ router.get('/:courseId/', async (req, res, next) => {
 		// ################################ end new Editor check ##################################
 		let ltiTools = [];
 		if (course.ltiToolIds && course.ltiToolIds.length > 0) {
-			ltiTools = await api(req).get('/ltiTools', {
-				qs: {
-					_id: { $in: course.ltiToolIds },
-				},
-			});
+			if (VERSION === 'v3') {
+				const ltiToolResponse = await api(req, { version: 'v3' }).get('/ltiTools');
+				ltiTools = (ltiToolResponse.data || []).filter((tool) => course.ltiToolIds.includes(tool._id));
```

Comment:
this filter should be moved into the backend. The easiest and cleanest would be to have a special endpoint to get the tools of a course

Reply:
/courses/:courseId/ltiTools